src/System.Web.HttpUtility/src/System/Web/Util/UriUtil.cs

```diff
@@ -50,13 +52,11 @@ internal static bool TrySplitUriForPathEncode(string input, out string schemeAnd
                     // To retain the same string as originally given, find the authority in the original url and include
                     // everything up to that.
                     int authorityIndex = inputWithoutQueryFragment.IndexOf(authority, StringComparison.OrdinalIgnoreCase);
-                    if (authorityIndex != -1)
-                    {
-                        int schemeAndAuthorityLength = authorityIndex + authority.Length;
-                        schemeAndAuthority = inputWithoutQueryFragment.Substring(0, schemeAndAuthorityLength);
-                        path = inputWithoutQueryFragment.Substring(schemeAndAuthorityLength);
-                        return true;
-                    }
+                    Debug.Assert(authorityIndex != -1); // Otherwise authority would be null or empty.
```

Comment:
Is it possible it might not be found if, for example, the input had unicode characters?  I'm not sure how Uri handles punycode with Authority.

Reply:
>I'm not sure how Uri handles punycode with Authority.

In terms of System.Uri, the "hostname" (authority) component will use either punnycode or UTF-8 depending on the IDN switch in the URI section of the app.config file.  However, for UWP apps and .NET Core which doesn't have that, it defaults to UTF-8.  But the bytes used for DNS resolution really uses OS settings for determining when to send IDN (punnycode) or UTF-8 on the wire for DNS resolution.  This is a complicated topic that I can't explain easily.

But I'm not sure how any of this maps to the HttpUtility class.  The bottom line here is that this code should behave the same as .NET Framework version of it.

Reply:
Thanks.  My real question is this:
Is it possible with the code:
```C#
var uri = new Uri(inputString);
bool contained = inputString.Contains(uri.Authority);
```
for `contained` to be false.  If it is possible, then this transformation is invalid.

Reply:
If the uri is a relative uri (and not an absolute uri), then the Authority portion can be missing.  I'm not sure though if this class supports that though.